Program.cs

```diff
@@ -5,11 +5,91 @@ namespace Calculator
   public static class Program
   {
     private static void Main(string[] args)
+    {
+      PrintWelcomeMessage();
+
+      while (true)
+      {
+        PerformOneCalculation();
+      }
+    }
+
+    private static void PrintWelcomeMessage()
     {
       Console.WriteLine("Welcome to the calculator!");
       Console.WriteLine("==========================");
+    }
+
+    private static void PerformOneCalculation()
+    {
+      var op = AskForOperator();
+      var n = AskForNumberArray(op);
+      var a = CalculateAnswer(op, n);
+
+      Console.WriteLine("The answer is: " + a);
+      Console.WriteLine();
+    }
+
+    private static string AskForOperator()
+    {
+      string[] allowedOperators = { "+", "-", "*", "/" };
+      Console.Write("Please enter the operator: ");
+      string op = Console.ReadLine();
+      return op;
+    }
+
+    private static int[] AskForNumberArray(string op)
+    {
+      Console.Write($"How many numbers do you want to {op}? ");
+
+      int count = int.Parse(Console.ReadLine());
+
+      int[] numbers = new int[count];
+      for (int index = 0; index < count; index++)
+      {
+        numbers[index] = AddNumber($"Please enter number {index + 1}: ");
+      }
+      return numbers;
+    }
+
+    private static int AddNumber(string message)
+    {
+//      Console.Write(message);
+
+      int number;
+
+      int.TryParse(Console.ReadLine(), out number);
+
+      return number;
+    }
+
+    public static int CalculateAnswer(string op, int[] numbers)
+    {
+      // Returns 0 if there aren't any numbers.
+      int answer = numbers[0];
+
+      int index = 0;
+      foreach (int number in numbers)
+      {
+        if (index == 0)
+        {
+          // Skip this one.
+        }
+        else
+        {
+          if (op == "/")
+            answer = answer / number;
+          else if (op == "+")
+            answer = answer + number;
+          else if (op == "-")
+            answer = answer - number;
+          else
+            Console.WriteLine("Invalid operator");
```

Comment:
Three things:
 - I think the `*` case has been missed here, please could you add it?
 - This might be a matter of taste but I think a `switch` expression might be a bit nicer here. It would look something like 
    ``` Csharp
    op switch 
    {
        "+" => answer += number;
        "-" => answer -= number;
        "*" => answer *= number;
        "/" => answer /= number;
        _   => Console.WriteLine("Invalid operator");
    } 
    ```
 - Another thing (that I haven't fixed in my example above) is that the "Invalid operator" will get printed `n-1` times for `n` numbers of user input. With more thorough validation as proposed above, this should never happen, but we could throw a custom exception so the error can be handled by the program. This lines up with your current `ThrowException` test that expects an exception for a different type of invalid input; an empty list of numbers, although I have suggested some changes for that above to throw a custom exception for that case in this method too.